Support log forwarding customisation for groups of namespaces (#28)
appuio/component-openshift4-logging: docs/modules/ROOT/pages/references/parameters.adoc

See the https://docs.openshift.com/container-platform/latest/logging/cluster-logging-external.html[OpenShift docs] for available parameters.

=== `clusterLogForwarding.namespace_groups`

[horizontal]
type:: dictionary
default:: {}

Customization for the logging of a specified group of namespaces.

> Enable forwarders will send the logs of the specified namespaces to a third-party log aggregator. Depending on the need you may need to deploy a separate log forwarder.

Reviewer: ```suggestion
Enabling forwarders will send the logs of the specified namespaces to a third-party log aggregator.
For some log aggregation systems you may need to deploy a separate log forwarder.
```

> Enable json parsing for a 'namespace_group' only makes sense if the logs are forwarded to the clusters default elasticsearch instance. Therefor 'default' will automatically be added to the forwarders.

Reviewer: This probably doesn't belong in this PR anymore

[source,yaml]
clusterLogForwarding:
  namespace_groups:
    my-group: <1>
      namespaces: <2>
        - my-namespace
      forwarders: <3>
        - splunk-forwarder
----

<1> Namespace to configure.
<2> List of namespaces.
<3> List of forwarders (defined in `clusterLogForwarding.forwarders`).

=== `clusterLogForwarding.application_logs`

[horizontal]
type:: dictionary
default:: {}

Customization for the logging of all applications.

> Enable forwarders will send the logs of all namespaces to a third-party log aggregator. Depending on the need you may need to deploy a separate log forwarder.

Reviewer: ```suggestion
Enabling forwarders will send the logs of all namespaces to a third-party log aggregator.
For some log aggregation systems you may need to deploy a separate log forwarder.
```
